app/src/main/java/com/shvants/runninglife/MainActivity.java

```diff
@@ -26,5 +47,63 @@ public void onClick(final View v) {
                         .commit();
             }
         });
+
+        serviceIntent = new Intent(this, CounterService.class);
+
+        findViewById(R.id.main_start_service_btn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
+                start = System.currentTimeMillis();
+                startService(serviceIntent);
+                bindService(serviceIntent, connection, BIND_AUTO_CREATE);
+            }
+        });
+
+        findViewById(R.id.main_stop_service_btn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View v) {
```

Comment:
Calling this logic if service is not started will lead to crash.

Reply:
@ajaxspace made stop button not clickable until start button is pressed https://github.com/AliakseiShvants/RunningLife/compare/01842e8f4dd29c265bdc458cc8a8a71b6eae8310...58e65bc2b5d572145ec44afbcbc683ee2b932035